packages/graphql-hooks/src/utils.ts

```diff
@@ -1,7 +1,39 @@
+import type {
+  DocumentNode,
+  OperationDefinitionNode
+} from 'graphql/language/ast'
+import { Kind } from 'graphql/language/kinds'
+import { print } from 'graphql/language/printer'
+
 /**
  * Pipe with support for async functions
  * @param {array} array of functions
  * @returns single function
  */
 export const pipeP = (fns: (() => any)[]) => (arg: any) =>
   fns.reduce((p, f) => p.then(f), Promise.resolve(arg))
+
+export function extractOperationName(document: DocumentNode | string): string | undefined {
```

Comment:
let's make sure that any new function we create has unit tests please, especially a function like this one that is 1. easy to unit test and 2. has a fairly large set of possible outcomes depending on the input

Reply:
Tests added.